src/odemis/gui/cont/streams.py

```diff
@@ -2734,42 +2767,47 @@ def addTemporalSpectrum(self):
             main_data.streak_delay,
             sstage=main_data.scan_stage,
             opm=self._main_data_model.opm,
+            axis_map=axes,
             detvas=detvas,
             streak_unit_vas=get_local_vas(main_data.streak_unit, self._main_data_model.hw_settings_config))
 
         # Create the equivalent MDStream
         sem_stream = self._tab_data_model.semStream
         sem_ts_stream = acqstream.SEMTemporalSpectrumMDStream("SEM TempSpec", [sem_stream, ts_stream])
 
-        spg = self._getAffectingSpectrograph(main_data.streak_ccd)
+        return self._addRepStream(ts_stream, sem_ts_stream)
+
+    def addMonochromator(self):
+        """ Create a Monochromator stream and add to to all compatible viewports """
+
+        main_data = self._main_data_model
+        spg = self._getAffectingSpectrograph(main_data.spectrometer)
 
-        # TODO: all the axes, including the filter band should be local. The
         # band should be set to the pass-through by default
```

Comment:
Please mark this comment as a `TODO`. (now that the axis is local, it should be easy to implement actually, but let's do it separately).